Drop log, move loader into optional options object

earldouglas opened the pull request:
Fixes #11.

earldouglas committed:
Drop log, move loader into optional options object

earldouglas commented on routeswitch.js:
```diff
@@ -216,20 +215,20 @@ function loadHandlers (loader, path, log) {
-                } else if (log) {
-                    log('error/handler', e, stat.name, e && e.stack);
+                } else {
+                    throw e;
```
This feels ugly, but it doesn't seem like we want to throw any exceptions caused when `stat` is a directory, so we drop that case on the floor and retry, and otherwise re-throw the exception.

gwicke replied:
The reason for trying the require first is that it might be a module on its own (with a package.json). We could test for that manually, but that would be even more complex.

earldouglas replied:
So this try/catch/rethrow is good to go then?

gwicke replied:
The alternative would be to test for the presence of package.json, check that it's actually json & has the right fields etc. (Okay, could perhaps just check for the file, but still..)

gwicke committed:
Drop node 0.8 support

The jshint version we now use is only available for node 0.8, so the travis
builds failed for 0.8.
